File: tourism-be/src/Modules/Stakeholders/Explorer.Stakeholders.Core/Domain/Followers/Follower.cs
using Explorer.BuildingBlocks.Core.Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Explorer.Stakeholders.Core.Domain.Followers
{
    public class Follower : Entity
    {
        public long FollowerId { get; init; }
        public long FollowedId { get; init; }
        public FollowerNotification Notification { get; init; }

        public Follower(long followerId, long followedId, FollowerNotification notification)
        {
            FollowerId = followerId;
            FollowedId = followedId;
            Notification = notification;

            Validate();
        }

        private void Validate()
        {
            if (FollowerId == 0) throw new ArgumentException("Invalid FollowerId");
            if (FollowedId == 0) throw new ArgumentException("Invalid FollowingId");
        }
    }
}

File: golang-be/Followers/model/Follower.go
/*package model

import (
	"errors"
)

type Follower struct {
	Id           int64 `gorm:"primaryKey"`
	FollowerID   int64
	FollowedID   int64
	Notification FollowerNotification `gorm:"embedded"`
}

func NewFollower(followerID, followedID int64, notification FollowerNotification) (*Follower, error) {
	if followerID == 0 {
		return nil, errors.New("invalid FollowerID")
	}
	if followedID == 0 {
		return nil, errors.New("invalid FollowedID")
	}

	return &Follower{
		FollowerID:   followerID,
		FollowedID:   followedID,
		Notification: notification,
	}, nil
}

//func (Follower) TableName() string {
//	return "Followers"
//}
*/

package model

import (
	"errors"
)

type Follower struct {
	FollowerID   int64                `json:"followerId" gorm:"primaryKey"`
	FollowedID   int64                `json:"followedId"`
	Notification FollowerNotification `json:"notification" gorm:"embedded"`
}

func NewFollower(followerID, followedID int64, notification FollowerNotification) (*Follower, error) {
	if followerID == 0 {
		return nil, errors.New("invalid FollowerID")
	}
	if followedID == 0 {
		return nil, errors.New("invalid FollowedID")
	}

	return &Follower{
		FollowerID:   followerID,
		FollowedID:   followedID,
		Notification: notification,
	}, nil
}
